test/rpc_test.py

```diff
@@ -1303,20 +1303,37 @@ def test_rref_context_debug_info(self):
         # barrier after check 3
         dist.barrier()
 
+    @dist_init
+    def test_disable_gil_profiling(self):
+        # test that rpc.enable_gil_profilig(false) will result in
+        # GIL wait time not being recorded.
+        from torch.distributed.rpc.api import _agent
+        # GIL profiling should be disabled by default.
+        dst_rank = (self.rank + 1) % self.world_size
+        rpc.rpc_sync("worker{}".format(dst_rank), torch.add, args=(torch.ones(1), torch.ones(1)))
+        info = _agent.get_debug_info()
+        self.assertRaises(KeyError, lambda: info["agent.gil_average_wait_time_us"])
+        rpc.enable_gil_profiling(True)
+        rpc.rpc_sync("worker{}".format(dst_rank), torch.add, args=(torch.ones(1), torch.ones(1)))
+        info = _agent.get_debug_info()
+        self.assertIn("agent.gil_average_wait_time_us", info)
+
     @dist_init
     @requires_process_group_agent("PROCESS_GROUP rpc backend specific test, skip")
     def test_process_group_debug_info(self):
```

Comment:
Filed https://github.com/pytorch/pytorch/issues/32362 to ensure that we eventually make this test backend agnostic.